port framework_agreement_sourcing to 8.0

lepistone opened the pull request:
Dependencies:
- [x] merge https://github.com/OCA/vertical-ngo/pull/8
- [x] merge https://github.com/OCA/purchase-workflow/pull/34

Checklist:
- [x] make tests pass in v8
- [x] lint
- [x] functional test

lepistone committed:
minimal work to make tests pass in v8

lepistone committed:
autopep8

lepistone committed:
manual lint

yvaucher commented:
Warning:

```
File ".../vertical-ngo/framework_agreement_sourcing/model/logistic_requisition_cost_estimate.py", line 93, in cost_estimate
      if x and x.logistic_requisition_source_id.procurement_method
  AttributeError: 'sale.order.line' object has no attribute 'logistic_requisition_source_id'
```

logistic_requisition_source_id was removed from sale.order

lepistone commented:
@yvaucher well spotted, you're right. That method needs to be adapted then, let's talk about it when you want. Thanks!

yvaucher commented on framework_agreement_sourcing/model/logistic_requisition_cost_estimate.py:
```diff
@@ -24,11 +24,11 @@
 
 
 class logistic_requisition_cost_estimate(orm.Model):
```
Should certainly be a TransientModel

lepistone committed:
bump version

lepistone committed:
typo in manifest

lepistone committed:
adapt tests to changes in the LR module

lepistone committed:
adapt to changes in models

Specifically, the field logistic_requisition_source_id is not there
anymore.

lepistone committed:
fix validation of a cost estimate

lepistone committed:
do not override function field for purchase line

It does not seem to make much difference from the original one in
logistic_requisition.

lepistone committed:
remove onchange agreement

The agreement field should already have a domain: no need to check with
an onchange that a consistent choice is made.

lepistone committed:
lint

lepistone committed:
cost estimate is a wizard: treat is as such

lepistone committed:
update copyright

lepistone commented:
Fixed as suggested by @yvaucher and rebased.

lepistone commented:
I checked "functional testing". More testing would help for sure, but I think we can merge it.

yvaucher commented:
@lepistone can you rebase to relaunch travis ?

lepistone committed:
factor out a constant that is just a little string

lepistone commented:
We're :green_heart: :green_apple:

yvaucher commented on framework_agreement_sourcing/__openerp__.py:
```diff
@@ -34,8 +34,8 @@
-In this case tender flow is byassed and confirmed PO will be generated
-when logistic requisition is confirmed.
+In this case the tender flow is bypassed and confirmed PO will be generated
+when the logistic requisition is confirmed.
```
In description bellow this comment:
s/generating/generated/
s/currences/currencies/

yvaucher commented:
appart from non blocking typo :+1:

lepistone committed:
fix typos in manifest

lepistone commented:
typo fixed.

yvaucher commented:
:+1: thx

gurneyalex commented:
this broke the build probably because depending on #14
